src/service/cid-impl.js

```diff
@@ -100,21 +120,54 @@ class Cid {
  * (Say Analytics provider X). It is unique per user, that purpose
  * and the AMP origin site.
  * @param {!Cid} cid
- * @param {string} externalCidScope
+ * @param {!GetCidDef} getCidStruct
  * @param {!Promise} persistenceConsent
  * @return {!Promise<?string>}
  */
-function getExternalCid(cid, externalCidScope, persistenceConsent) {
+function getExternalCid(cid, getCidStruct, persistenceConsent) {
   const url = parseUrl(cid.win.location.href);
   if (!isProxyOrigin(url)) {
-    return Promise.resolve(getCookie(cid.win, externalCidScope));
+    return getOrCreateCookie(cid, getCidStruct, persistenceConsent);
   }
   return getBaseCid(cid, persistenceConsent).then(baseCid => {
     return cid.sha384Base64_(
         baseCid +
         getSourceOrigin(url) +
-        externalCidScope);
+        getCidStruct.scope);
+  });
+}
+
+/**
+ * If cookie exists it's returned immediately. Otherwise, if instructed, the
+ * new cookie is created.
+ *
+ * @param {!Cid} cid
+ * @param {!GetCidDef} getCidStruct
+ * @param {!Promise} persistenceConsent
+ * @return {!Promise<?string>}
+ */
+function getOrCreateCookie(cid, getCidStruct, persistenceConsent) {
+  const win = cid.win;
+  const existingCookie = getCookie(win, getCidStruct.scope);
+  if (existingCookie || !getCidStruct.createCookieIfNotPresent) {
+    return Promise.resolve(existingCookie);
+  }
+
+  // Create new cookie, always prefixed with "amp-", so that we can see from
+  // the value whether we created it.
+  const newCookie = 'amp-' + cid.sha384Base64_(getEntropy(win));
```

Comment:
My thinking is that the one we use in local storage has to be unique so separate websites cannot determine this is the same user. Here we just need a sentinel value, right?